Bg attach

simeydotme opened the pull request:
Hi @crutchcorn 

This is a little later than anticipated. Life and Hurricanes, lol.

So a couple of things;

1) updated all the CSS to utilise [what was in my Codepen](https://codepen.io/simeydotme/pen/azdzoQE), but kept your original css variable names / colors / class names.
2) Added dark mode
3) Now works in Firefox _(not sure if you realised your original css was not functional in firefox? it was related to the css type coercion)_
4) Modified the .README to include customisation and dark mode information
5) Added a `.svelte` example
6) Removed the core observer imports from the examples directories

---

On top of that, I have left some things out which I hope you can handle as it's more to do with direction/repo-management

1) I didn't change/delete anything with regards to the core observer package
2) I'm not super-familiar with monorepos, so I hope I didn't break anything (I was able to build, test and run locally)
3) I used my _(admittedly complex)_ css gradient, [which is a eased-gradient](https://larsenwork.com/easing-gradients/) and looks proper smooth ... but totally cool if you want to revert back to a simple linear gradient

---

And finally, there's one thing I want to highlight as a potential issue/improvement.

Personally, for performance and user-distraction reasons, I would never want to have hundreds of placeholders animating on a page at the same time. So typically I would implement a intersection observer to disable animating any placeholders not on the screen. But there's an issue (both with yours, and mine) that if an animation is paused, it becomes out of sync and when it is resumed it no longer maintains the illusion of a full-page-wipe. [Demo here; https://codepen.io/simeydotme/pen/YPwwqKw/5d3691f4607ac9b878e491f658808303 - scroll down and back up](https://codepen.io/simeydotme/pen/YPwwqKw/5d3691f4607ac9b878e491f658808303)

I had a thought about how to combat this; move the animation from each skeleton up to the root and having all skeletons inherit the `--skeleton-position` variable from `:root`, that way every element always has a single position instead of animating their own positions.

I think this might be better for performance, also _(1 animation instead of N)_... except that the animation would be running continuously no matter if a skeleton exists... so in that case I think it would be mandatory for an intersection observer to check if any loading elements are in the viewport, and stop/start the global `:root` animation. This is obviously a bit more complicated, and requires still a `use:` action/directive .. hence why I didn't fully delete that yet.

I'll let you think it over

simeydotme committed:
modify the CSS to match @simeydotme styling from codepen

- uses `background-attachment: fixed;` to style the background for all
elements
- added in a dark mode
- gradient changed to a smooth eased gradient for beauty
  - _(feel free to revert this to a simple one)_
- update the README to explain how to customise the variables

simeydotme committed:
update examples and packages, add svelte example

simeydotme committed:
add keywords for npm/github findability

crutchcorn commented:
> So typically I would implement a intersection observer to disable animating any placeholders not on the screen. 

I actually think this should not be part of this package. I think I want to lean towards removing all JS code and even the other non-HTML examples (I'll do so after we merge this PR).

> I had a thought about how to combat this; move the animation from each skeleton up to the root and having all skeletons inherit the --skeleton-position variable from :root

On the other hand, I think this is a good call and something we should tackle. Wanna change that and we can merge?

simeydotme commented:
Awesome ok I'll update it ;)

simeydotme committed:
move animation to the :root element

- fix placement of css variables to allow overriding colors at
element-level
- update Card/App components to show dark-mode and custom coloring
- move animation to `:root` so all animations are synced
	- remove `animation-repeat` variable as it doesn't make sense now
- simplify css gradient for performance gains
- update README to reflect
	- add a dark-mode animation example

simeydotme committed:
fix linting with prettier

crutchcorn committed:
feat: remove adapters, turning package into CSS only

autofix-ci[bot] committed:
ci: apply automated fixes and generate docs

crutchcorn commented:
@simeydotme you're amazing! Thank you so much for the help here; check your email
